File: win-prog/ProgWinCS/Brushes and Pens/BouncingGradientBrushBall/BouncingGradientBrushBall.cs
//--------------------------------------------------------
// BouncingGradientBrushBall.cs © 2001 by Charles Petzold
//--------------------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class BouncingGradientBrushBall: Bounce
{
	public new static void Main()
	{
		Application.Run(new BouncingGradientBrushBall());
	}
	public BouncingGradientBrushBall()
	{
		Text = "Bouncing Gradient Brush Ball";
	}
	protected override void DrawBall(Graphics grfx, Rectangle rect)
	{
		GraphicsPath path = new GraphicsPath();
		path.AddEllipse(rect);

		PathGradientBrush pgbrush = new PathGradientBrush(path);
		pgbrush.CenterPoint = new PointF((rect.Left + rect.Right) / 3,
			(rect.Top + rect.Bottom) / 3);
		pgbrush.CenterColor = Color.White;
		pgbrush.SurroundColors = new Color[] { Color.Red };

		grfx.FillRectangle(pgbrush, rect);
	}
}

File: win-prog/ProgWinVB/Brushes and Pens/BouncingGradientBrushBall/BouncingGradientBrushBall.vb
'----------------------------------------------------------
' BouncingGradientBrushBall.vb (c) 2002 by Charles Petzold
'----------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class BouncingGradientBrushBall
    Inherits Bounce

    Shared Shadows Sub Main()
        Application.Run(New BouncingGradientBrushBall())
    End Sub

    Sub New()
        Text = "Bouncing Gradient Brush Ball"
    End Sub

    Protected Overrides Sub DrawBall(ByVal grfx As Graphics, _
                                     ByVal rect As Rectangle)
        Dim path As New GraphicsPath()
        path.AddEllipse(rect)

        Dim pgbr As New PathGradientBrush(path)
        pgbr.CenterPoint = New PointF((rect.Left + rect.Right) \ 3, _
                                      (rect.Top + rect.Bottom) \ 3)
        pgbr.CenterColor = Color.White
        pgbr.SurroundColors = New Color() {Color.Red}
        grfx.FillRectangle(pgbr, rect)
    End Sub
End Class
